### PR TITLE
IL UP Receiver migration Pre-Work

### PR DESCRIPTION
**Pull request by oslynn**
This PR IL receiver UP migration pre-work,

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/il-phd.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# To submit this to localhost, run:
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/il-phd.yml:1]: same comment as on ND. We shouldn't be storing the receiver settings here anymore as it is messing up platform's new feature to automatically check transforms.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/il-phd.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        - "(Bundle.entry.resource.ofType(ServiceRequest)[0].requester.resolve().organization.resolve().address.state = 'IL') or (Bundle.entry.resource.ofType(Patient).address.state = 'IL')"
+      qualityFilter: []
+      routingFilter:
+        - "((Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or Bundle.entry.resource.ofType(Specimen).collection.collected.exists()) or Bundle.entry.resource.ofType(serviceRequest).occurrence.exists() or Bundle.entry.resource.ofType(Observation).effective.exists())"
+        - "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or Bundle.entry.resource.ofType(Patient).telecom.exists())"
+        - "Bundle.entry.resource.ofType(MessageHeader).id.exists()"
+        - "Bundle.entry.resource.ofType(Patient).birthDate.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.family.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.given.count() > 0"
+        - "Bundle.entry.resource.ofType(Specimen).type.exists()"
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/il-phd.yml:75]: Move to qualityFilter.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/IL-receiver-transform.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    condition: 'true'
+    value: [ '"CLIA"' ]
+    hl7Spec: [ 'MSH-4-3' ]
+
+  - name: il-name-of-the-receiving-application
+    condition: 'true'
+    value: [ '""' ]
+    hl7Spec: [ 'MSH-5-1' ]
+
+  - name: il-OID-of-the-receiving-application
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/IL-receiver-transform.yml:37]: Why are we suppressing all fields from MSH-5 and MSH-6? I don't see that in their existing settings.

**Review by victor-chaparro — CHANGES_REQUESTED** [review]
I see they have this setting: `"suppressNonNPI": true,` in the covid pipeline. I don't think we've seen this setting before? Do you know what it's doing? We probably need to add that as a receiver transform.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/IL-receiver-transform.yml (hunk 3 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IL/IL-receiver-transform.yml:37]: Here is the setting above.        
        receivingApplicationName: null
        receivingApplicationOID: null
        receivingFacilityName: null
        receivingFacilityOID: null

File: prime-router/settings/STLTs/IL/il-phd.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        replaceUnicodeWithAscii: false
+        useBlankInsteadOfUnknown: null
+        truncateHDNamespaceIds: true
+        truncateHl7Fields: "OBX-23-1, ORC-21-1"
+        usePid14ForPatientEmail: false
+        convertTimestampToDateTime: null
+        cliaForOutOfStateTesting: null
+        cliaForSender: {}
+        phoneNumberFormatting: "STANDARD"
+        suppressNonNPI: true
```
Comment (oslynn) [inline prime-router/settings/STLTs/IL/il-phd.yml:45]: Need to suppressNonNPI.

File: prime-router/settings/STLTs/IL/il-phd.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        whenEmpty:
+          action: "NONE"
+          onlyOncePerDay: false
+      description: ""
+      transport: !<SFTP>
+        host: "moveit.illinois.gov"
+        port: "22"
+        filePath: "/Distribution/DPH/LabData/HL7-251/"
+        credentialName: null
+        type: "SFTP"
```
Comment (victor-chaparro) [inline prime-router/settings/STLTs/IL/il-phd.yml:101]: Remove transport

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

**Pull request merged**